app/models/historical_event.rb

```diff
@@ -1,6 +1,7 @@
 class HistoricalEvent < ApplicationRecord
   belongs_to :user
   belongs_to :activity
+  belongs_to :report, optional: true
```

Comment:
I think making this optional for now is fine, especially given the fact that it'll make tests fail on the imports. We can revisit this later if needs be

Reply:
I've tweaked the integration test to make the assertion about the identity of the associated report more precise. In the case of the 'BEIS user' scenario, there is no report. I've added a note in https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/1209/commits/69b861a8d62998ffc30eef26429067eb3d27414d as I'm not sure what that signifies... Maybe @mec will have a view?

Reply:
Ah yeah, good call - BEIS users don't need a report to exist to make any changes, so it probably makes sense for the Report to be optional in this case

Reply:
Hey! Yep, totally agree, the report association has to be optional as BEIS users will make changes to their activites without a report.

Reply:
thanks both, we can figure out how to make it a bit more explicit in the code/specs. Can we say:

- when using wizard: report is mandatory for DP and optional for BEIS
- when using upload CSV: report is always mandatory (it's a URL param after all!)
- when using 'rake' import facility: report is always optional

Reply:
- when using wizard: report is mandatory for DP and optional for BEIS: yep
- when using upload CSV: report is always mandatory (it's a URL param after all!): we will probably need it to be optional as BEIS will need to upload data as well, but we are not there yet.
- when using 'rake' import facility: report is always optional: as it's optional dependant on the user I think it would be the same here.